### ISSUE TITLE
Support for linuxOSConfig on azurerm_kubernetes_cluster_node_pool

### ISSUE THREAD

**Issue by liammoat** [terraform-providers/terraform-provider-azurerm#10038]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In API version 2020-11-01 of the Azure Rest API `linuxOSConfig` can be used to configured sysctl properties on an AKS agent pool (see [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-11-01/examples/AgentPoolsCreate_CustomNodeConfig.json)).

To help maintain feature parity, it would make sense to introduce this API in the Terraform provider. The API is available in ARM (see [here](https://docs.microsoft.com/en-us/azure/templates/microsoft.containerservice/managedclusters/agentpools)).

### New or Affected Resource(s)

* azurerm_kubernetes_cluster_node_pool

### Potential Terraform Configuration

```hcl
resource "azurerm_kubernetes_cluster_node_pool" "example" {
  name                  = "internal"
  kubernetes_cluster_id = azurerm_kubernetes_cluster.example.id
  vm_size               = "Standard_DS2_v2"
  node_count            = 1

  linux_os_config {
    sysctls {
      netIpv4TcpWmem = 12345
      netIpv4TcpTwReuse = true
      netIpv4IpLocalPortRange = "20000 60000"
      kernelThreadsMax = 99999
    }
  }
}
```

### References

* https://docs.microsoft.com/en-us/azure/aks/custom-node-configuration
* https://docs.microsoft.com/en-us/azure/templates/microsoft.containerservice/managedclusters/agentpools

**Comment by liammoat** [comment]
cc: @grayzu

**Comment by katbyte** [comment]
closing as it looks like this has been fixed by #11119
